src/lora_plan.erl

```diff
@@ -0,0 +1,848 @@
+-module(lora_plan).
+
+-export([
+    freq_to_channel/2,
+    channel_to_freq/2,
+    tx_power/2,
+    tx_power_list/1,
+    tx_power_table/1,
+    region_to_plan/1,
+    rx2_datarate/1,
+    rx2_tuple/1,
+    max_uplink_payload_size/2,
+    max_downlink_payload_size/2,
+    max_payload_size/2,
+    datarate_to_index/2,
+    index_to_datarate/2,
+    datar_to_down/3,
+    dr_to_down/3,
+    datarate_to_atom/1,
+    atom_to_datarate/1,
+    downlink_eirp/2,
+    max_uplink_snr/1,
+    rx_window/5,
+    join1_window/3,
+    join2_window/2,
+    rx1_window/4,
+    rx2_window/3,
+    rx1_or_rx2_window/4
+]).
+
+-include("lora.hrl").
+
+-spec region_to_plan(atom()) -> #channel_plan{}.
+region_to_plan(Region) ->
+    case Region of
+        'EU868' -> plan_eu868();
+        'US915' -> plan_us915();
+        'AU915' -> plan_au915();
+        'CN470' -> plan_cn470();
+        'KR920' -> plan_kr920();
+        'IN865' -> plan_in865();
+        'AS923' -> plan_as923_1();
+        'AS923_1' -> plan_as923_1();
+        'AS923_2' -> plan_as923_1();
+        'AS923_3' -> plan_as923_1();
+        'AS923_4' -> plan_as923_1()
+    end.
+
+-spec datarate_to_atom(binary()) -> atom().
+datarate_to_atom(Binary) ->
+    case Binary of
+        <<"SF12BW125">> -> 'SF12BW125';
```

Comment:
Maybe you're already aware: there's `erlang:binary_to_existing_atom()`
which returns `badarg` upon no match that may be used for the catch-all `RFU`.  These atoms already exist elsewhere; e.g., line 432.